### DISCUSSION TITLE
Every nth day of the week of the month

### DISCUSSION THREAD

**Discussion by bbailey1024** [go-co-op/gocron#273] [Q&A]
Is there a simple mechanism for scheduling this? For example, run on the 1st Tuesday of every month.

Counting weeks from a start date is only an approximation of intent and doesn't meet the requirement. Same goes for specifying a day of the month. In my use case the scheduler and function need to be independent so scheduling logic can't be included in the function. 
Perhaps there's a combination of scheduling tactics I'm missing that could achieve this goal. Any tips would be helpful. 

Thanks!

**Comment by JohnRoesler** [comment]
Hi @bbailey1024 ! 

There is not currently a way with the DSL to schedule every 1st Tuesday of the Month. We should add that! 

For specifying a specific day or days of the Month that can be done with https://pkg.go.dev/github.com/go-co-op/gocron#example-Scheduler.Month

Yes of course if you were okay combing logic you could schedule weekly on Tuesdays and check the date, but I understand the need to keep these separate.

Unfortunately this isn't something you can do with a cron statement alone, so the Cron method won't solve it here either.

**Comment by bbailey1024** [comment]
Thanks for the feedback and I figured that might be the answer. It's just an odd bit of scheduling logic that doesn't fit well into the very well defined methods this library offers, but something a client will invariably ask for I'm sure. I'll probably end up writing a function to derive the days associated with the request and create single jobs for each of those days. Perhaps tag them all with a unique id to keep them grouped. If that code doesn't come out looking too hacky I can post it here.

**Reply by JohnRoesler** [reply]
I would be interested to see what you come up with!

**Reply by bbailey1024** [reply]
I spent some time coming up with the logic for generating timestamps based on the nth day of the month logic. End result is a list of time.Time that I thought to manually schedule with gocron, but looks like you can't create an arbitrary RunOnce style job.

Getting "next month's" run time could be generated pretty easily from the attached code I think. Maybe it could fit into something like Every(1).Month().NthWeekday(time.Time).

In the meantime, if there's a way to schedule run once style jobs that'd help me use this code to get the job done. Then I could walk it back if you decide to implement something like I mentioned above. I saw https://github.com/go-co-op/gocron/issues/274, but I don't think I have the time to contribute directly unfortunately.

https://gist.github.com/bbailey1024/1aa7205bd61220cba8b9661a6315d148

Let me know what you think.

**Reply by bbailey1024** [reply]
Closing the loop in case anyone is interested or finds this helpful. I updated the above gist to implement run once style jobs via the LimitRunsTo method. Not sure how I missed that on the first pass, but there it is.

**Comment by aphsa** [comment]
I have a similar request. Do we have a solution to run Every 2nd Monday of the month?
